### DISCUSSION TITLE
What is React Js?

### DISCUSSION THREAD

**Discussion by gokarna123-goku** [gokarna123-goku/try_version#2] [Q&A] [answered]
React is a free and open-source front-end JavaScript library for building user interfaces based on components. It is maintained by Meta and a community of individual developers and companies.

**Comment by gokarna123-goku** [comment, marked as answer]
React.js, more commonly known as React, is a free, open-source JavaScript library. It works best to build user interfaces by combining sections of code (components) into full websites. Originally built by Facebook, Meta and the open-source community now maintain it.

**Comment by gtech-official08** [comment]
I totally agree with his answer

**Comment by gokarna123-goku** [comment]
React.js is built using JSX – A combination of JavaScript and XML. Elements are created using JSX, then use JavaScript to render them on your site. While React has a steep learning curve for a junior developer, it’s quickly shaping into one of the most popular and in-demand JavaScript libraries.

**Comment by gokarna123-goku** [comment]
React is considered a JavaScript library rather than a framework, whereas the other options we’ll consider today are considered frameworks. It helps to think of a library as a tool that developers could use in any project and a framework as a whole design.

**Comment by gokarna123-goku** [comment]
The React.js framework is an open-source JavaScript framework and library developed by Facebook. It’s used for building interactive user interfaces and web applications quickly and efficiently with significantly less code than you would with vanilla JavaScript.
